apps/server/src/services/ElasticSearchClient.ts

```diff
@@ -48,24 +49,39 @@ export async function reIndexElasticSearch() {
   } catch {
     info('Elasticsearch "institutions" index did not exist')
   }
-  info('Elasticsearch indexing institutions')
-  const dataFilePath = getInstitutionFilePath()
-  const rawData = readFileSync(dataFilePath)
-  const jsonData = JSON.parse(rawData.toString())
+  const institutionData = await getCachedInstitutions()
+  const insIds = institutionData.map((ins: CachedInstitution) => ins.ucp_id)
+
+  await overwriteSet(INSTITUTION_CURRENT_LIST_IDS, insIds)
 
   await ElasticsearchClient.indices.create({ index: 'institutions' })
 
-  const indexPromises = jsonData.map(async (institution: { ucp_id: any }) => {
-    return await ElasticsearchClient.index({
-      index: 'institutions',
-      id: institution.ucp_id,
-      document: institution
-    })
-  })
+  const indexPromises = institutionData.map(
+    async (institution: { ucp_id: any }) => {
+      return await ElasticsearchClient.index({
+        index: 'institutions',
+        id: institution.ucp_id,
+        document: institution
+      })
+    }
+  )
 
   await Promise.all(indexPromises)
 }
 
+async function getCachedInstitutions(): Promise<CachedInstitution[]> {
+  const response = await getCachedInstitutionListFromServer()
```

Comment:
We check on the scheduled sync to make sure that there are institutions before we choose to use the response from the server, but we're not checking here